include/deal.II/matrix_free/fe_point_evaluation.h

```diff
@@ -874,7 +944,8 @@ FEPointEvaluation<n_components, dim, spacedim, Number>::evaluate(
   const bool precomputed_mapping = mapping_info_on_the_fly.get() == nullptr;
   if (precomputed_mapping)
     {
-      unit_points = mapping_info->get_unit_points();
+      unit_points =
+        mapping_info->get_unit_points(current_cell_index, current_face_number);
```

Comment:
How do I check this? We could get rid of `unit_points` in `FEPointEvaluation` entirely and always ask `MappingInfo` (maybe in a follow-up).

Reply:
If we could move all code into `MappingInfo` (also the case where the user never sets up such an object explicitly, e.g. in particle codes that want to evaluate information once), that would be great. Regarding how to check it: I use a compiler/trace tool and monitor the allocations, first using toy codes, to see how the compiler reacts. Here I use this one:
```
#include <vector>
#include <iostream>

int main()
{
  std::vector<unsigned int> data(5, 3);
  std::vector<unsigned int> second(3, 6);
  for (unsigned int i=0; i<10; ++i)
    data = second;
  std::cout << data.size() << " " << data.capacity() << " " << second.size() << " " << second.capacity() << std::endl;
}
```
Using valgrind as `valgrind ./a.out` to check for memory errors, I now see 
```
==269622==   total heap usage: 4 allocs, 4 frees, 73,760 bytes allocated
```
This is less than 10 repetitions, so it can't allocate memory inside the loop (one might wonder if compilers might optimize it away, as it does not change the result, but at least gcc does not here). The reason for the loop is that I see more than 2 allocs, which I would expect for the two vectors - gcc counts the full program here. The cause is that `std::cout << ` does another one to put the data to the output stream, and the last one is in the global program `init` outside main (it is done inside `call_init.part.0` and part of the program startup, outside the control of my program).
The output is also revealing, which is why I included it:
```
3 5 3 3
```
You see from the first two numbers that the vector `data` kept its original capacity of 5, even though the length was changed to 3 after assignment from `second`. This is a strong indication that it kept its memory region. You could output the address of `&data[0]` to ensure it kept the same. You can also look at the source code of `stl_vector.h` to see this (for me, the file is `/usr/include/c++/12/bits/stl_vector.h`, but the implementation is in `)/usr/include/c++/12/bits/vector.tcc`:
```
  template<typename _Tp, typename _Alloc>
    _GLIBCXX20_CONSTEXPR
    vector<_Tp, _Alloc>&
    vector<_Tp, _Alloc>::
    operator=(const vector<_Tp, _Alloc>& __x)
    {
       ...
          const size_type __xlen = __x.size();
       ...
          else if (size() >= __xlen)
            {
              std::_Destroy(std::copy(__x.begin(), __x.end(), begin()),
                            end(), _M_get_Tp_allocator());
            }
       ...
```
From this, you see that we only call `std::copy` (and destroy the elements that now got invalid due to reducing the size), but we did not deallocate/reallocate.